recipes-support/lvm2/lvm2-conf-initramfs_1.0.bb

```diff
@@ -7,6 +7,9 @@ LIC_FILES_CHKSUM = "file://${COMMON_LICENSE_DIR}/MIT;md5=0835ade698e0bcf8506ecda
 SRC_URI = "file://lvm.conf"
 
 inherit allarch
+inherit multilib_script
+
+MULTILIB_SCRIPTS = "${PN}:${sysconfdir}/lvm/lvm.conf"
```

Comment:
This change seems unexpected.  I'd be interested to see the actual error message (and maybe it should be in the commit message).

I'm surprised this recipe is a problem since it is marked allarch.  When I look up MULTILIB_SCRIPTS, it says 

> "Whilst the package managers handle multilib ELF binaries well, they don't handle scripts in the *bindir directories well. This adds support for marking these up so that they can be handled using update-alternatives."

This is a conf file and not a script/ *bindir file.  The recipe isn't using update-alternatives since it's supposed to override the lvm2 RRECOMMENDS.  A different package is installed which populates the same file system name.

Oh, is the issue that since lvm2 is multilib compatible, it's looking for package `lvm2-conf-${MULTILIB_SUFFIX}`, but this recipe (lvm2-conf-initramfs) isn't providing that name?  The MULTILIB_SCRIPT takes care of that?

Reply:
Could it be that "${PN}-conf" should be replaced with a fixed string here: https://github.com/OpenXT/xenclient-oe/blob/d24250170500f42bdffb7e02a7361e03d0ccb05f/recipes-support/lvm2/lvm2_%25.bbappend#L25 ?

That may prevent multilib from changing the variable? (If that is the issue).

Reply:
I need to dig into this to refresh my understanding of it and will either update the commit message or replace the commit, depending on what I find. I have observed that the upstream lvm2 recipe inherits the `multilib_script` class and applies `MULTILIB_SCRIPTS` to the `lvm.conf` file, here:

https://git.openembedded.org/meta-openembedded/tree/meta-oe/recipes-support/lvm2/lvm2_2.03.02.bb?h=zeus&id=663493e98883ada2a13a6e873700ed0eda654bb3#n13

and the change in this commit is mirroring that into the `lvm2-config-initramfs` recipe.

The error observed without this change is:

```
     WARNING: xenclient-initramfs-image-1.0-r0 do_rootfs:
        lvm2.postinst returned 1, marking as unpacked only, configuration
        required on target.
    
     ERROR: xenclient-initramfs-image-1.0-r0 do_rootfs: Postinstall
        scriptlets of ['lvm2'] have failed. If the intention is to defer
        them to first boot, then please place them into
        pkg_postinst_ontarget_${PN} ().
        Deferring to first boot via 'exit 1' is no longer supported.
```

and the contents of the script that fails are:

```
        #!/bin/sh
        set -e
            update-alternatives --install /etc/lvm/lvm.conf lvm.conf /etc/lvm/lvm.conf-lib 10
        if [ x"$D" = "x" ]; then
            if [ -x /sbin/ldconfig ]; then /sbin/ldconfig ; fi
        fi
```

Reply:
If you put the error message in the commit message, and the part about "lvm2 recipe inherits the multilib_script class and applies MULTILIB_SCRIPTS to the lvm.conf file", I think that is fine.  A comment in the file would be good, too since it's not obvious why its there without looking at the changelog.  Don't spend too much time on this.  Thanks.